### ISSUE TITLE
Sweeps with wandb 0.10.8 keyerror

### ISSUE THREAD

**Issue by Odrec** [wandb/client#1408]
After running around 3 or 4 sweeps with some models from simpletransformers I am getting this error

```
Traceback (most recent call last):
  File "main.py", line 204, in <module>
    SM.main()
  File "/root/semantic_models.py", line 177, in main
    wandb.agent(sweep_id, function=lambda: self.train(train_args))
  File "/root/complex_semantics/lib/python3.8/site-packages/wandb/wandb_agent.py", line 520, in agent
    return pyagent(sweep_id, function, entity, project, count)
  File "/root/complex_semantics/lib/python3.8/site-packages/wandb/agents/pyagent.py", line 316, in pyagent
    agent.run()
  File "/root/complex_semantics/lib/python3.8/site-packages/wandb/agents/pyagent.py", line 298, in run
    self._run_jobs_from_queue()
  File "/root/complex_semantics/lib/python3.8/site-packages/wandb/agents/pyagent.py", line 251, in _run_jobs_from_queue
    raise e
  File "/root/complex_semantics/lib/python3.8/site-packages/wandb/agents/pyagent.py", line 235, in _run_jobs_from_queue
    del self._run_threads[job.run_id]
KeyError: 'jb9k4gpr'
```

**Comment by cvphelps** [comment]
Thanks for reporting this. My colleague @farizrahman4u is taking a look

**Comment by raubitsj** [comment]
This is fixed in 0.10.9
